### PR TITLE
Actually install the package inside the Dockerfile so the UI works

### PR DESCRIPTION
**Pull request by mdagost**
I saw a Dockerfile and assumed that would be the easiest way to get up-and-running with development, but it turns out there was more work involved.  This PR actually installs the dependencies and the package and compiles the js so the UI works.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -5,3 +5,11 @@ RUN pip install numpy pandas flask pygal smalluuid zipstream python-dateutil git
 WORKDIR /app
 
 ADD . /app
+
+RUN pip install -r dev-requirements.txt
```
Comment (andrewmchen) [inline Dockerfile]: Can we remove line 3 here?

Comment (andrewmchen) [inline Dockerfile]: Could we collapse all of these RUN layers into one?
```
RUN pip install -r dev-requirements.txt \
    && pip install -r tox-requirements.txt \
    && ...
```

This way we can minimize the number of layers in our image. https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#minimize-the-number-of-layers

File: Dockerfile [hunk 2]
```diff
@@ -5,3 +5,11 @@ RUN pip install numpy pandas flask pygal smalluuid zipstream python-dateutil git
...
+RUN pip install -r tox-requirements.txt
+RUN pip install -e .
+
+RUN apt-get install -y gnupg && curl -sL https://deb.nodesource.com/setup_10.x | bash -
+RUN apt-get install -y nodejs
+RUN cd mlflow/server/js && npm run build
```
Comment (andrewmchen) [inline Dockerfile]: I ran this and it didn't work for me, do we need to run an `npm install` before this?

**Review by andrewmchen — COMMENTED** [review]
Thanks for the pull request! We actually accidentally left this in the repository but thank you for fleshing it out. I left some minor comments.

File: Dockerfile (hunk 1 above)
Comment (mdagost) [reply Dockerfile]: Ah yeah, that's not needed anymore.  Removed.

Comment (mdagost) [reply Dockerfile]: Done.

File: Dockerfile (hunk 2 above)
Comment (mdagost) [reply Dockerfile]: Yeah, good catch.  Thanks.

File: Dockerfile [hunk 3]
```diff
@@ -1,7 +1,11 @@
 FROM continuumio/miniconda
 
-RUN pip install numpy pandas flask pygal smalluuid zipstream python-dateutil gitpython scikit-learn
-
 WORKDIR /app
 
 ADD . /app
+
+RUN pip install -r dev-requirements.txt && pip install -r tox-requirements.txt && pip install -e .
```
Comment (andrewmchen) [inline Dockerfile]: Can we actually collapse all of these RUN's to a single RUN?

Also a nit on the `&&`, can we have each `&&` on the new line?

**Review by andrewmchen — COMMENTED** [review]
The linter is broken! This PR should fix https://github.com/databricks/mlflow/pull/19.

**Comment by mdagost** [comment]
@andrewmchen No worries on the nit.  Better now?

**Comment by andrewmchen** [comment]
LGTM now :) thanks for your contribution.

**Pull request merged**